uaclient/clouds/gcp.py

```diff
@@ -42,11 +42,34 @@ def __init__(self):
     # mypy does not handle @property around inner decorators
     # https://github.com/python/mypy/issues/1362
     @property  # type: ignore
-    @util.retry(HTTPError, retry_sleeps=[1, 2, 5])
+    @util.retry(
+        exceptions.GCPProServiceAccountNotEnabledError, retry_sleeps=[1, 2, 5]
+    )
     def identity_doc(self) -> Dict[str, Any]:
-        url_response, _headers = util.readurl(
-            TOKEN_URL, headers={"Metadata-Flavor": "Google"}
-        )
+        try:
+            headers = {"Metadata-Flavor": "Google"}
+            url_response, _headers = util.readurl(TOKEN_URL, headers=headers)
+        except URLError as e:
+            body = None
+            if hasattr(e, "body"):
+                body = getattr(e, "body", None)
+            elif hasattr(e, "read"):
+                reader = getattr(e, "read")()
+                body = reader.decode("utf-8")
+            error_desc = None
+            if body:
+                try:
+                    error_details = json.loads(
+                        body, cls=util.DatetimeAwareJSONDecoder
+                    )
+                except ValueError:
+                    error_details = None
+                if error_details:
+                    error_desc = error_details.get("error_description", None)
+            msg = error_desc if error_desc else getattr(e, "msg", None)
+            raise exceptions.GCPProServiceAccountNotEnabledError(
```

Comment:
I think we can create the whole error message directly here. Even the part that checks the error code. My rationale is that if we do that, we will not need to do anything on the cli module, since it already raises the `UserFacingError` and the `main_error_handler` decorator should already print the error message as we want

Reply:
Additionally, it seems we have an extra newline on the message:
```
Failed to attach machine
Service account not enabled on this instance.

For more information, see https://cloud.google.com/iam/docs/service-accounts
```

Reply:
Makes sense, because we will start having proliferation of instances of UserFacingError in cli.py.

Reply:
As a result, I think I will rename the exceptions class to GCPProAccountError to use for any future error. What do you think?